docs/data/tree-view/rich-tree-view/lazy-loading/BasicLazyLoading.js

```diff
@@ -0,0 +1,33 @@
+import * as React from 'react';
+import Box from '@mui/material/Box';
+import { randomInt, randomName, randomId } from '@mui/x-data-grid-generator';
+import { RichTreeView } from '@mui/x-tree-view/RichTreeView';
+
+const fetchData = async () => {
+  const rows = Array.from({ length: 10 }, () => ({
+    id: randomId(),
+    label: randomName({}, {}),
+    ...(randomInt(0, 1) ? { childrenCount: 10 } : {}),
+  }));
+
+  return new Promise((resolve) => {
+    setTimeout(() => {
+      resolve(rows);
+    }, 1000);
+  });
+};
+
+export default function BasicLazyLoading() {
+  return (
+    <Box sx={{ width: '300px' }}>
+      <RichTreeView
+        items={[]}
+        experimentalFeatures={{ lazyLoading: true }}
+        treeViewDataSource={{
+          getChildrenCount: (item) => item?.childrenCount,
+          getTreeItems: fetchData,
+        }}
```

Comment:
Not super clear to me if:

1. This should be part of MUI X and not Toolpad. But maybe https://github.com/mui/mui-x/issues/14731 is another layer of abstraction, on top of this one
2. How much code should be shared with the data grid

Reply:
For now we are following a DX that is close to the one of the grid.
Once we have an initial version fully working, we will see with the grid and the charts if there are things to share with the grid. Sharing DX is obvious (to me at least), sharing code might be more complex since it's very tight to the internals of each set of components.